crates/core/src/dht/pending_validations.rs

```diff
@@ -157,3 +166,36 @@ impl From<PendingValidationStruct> for EntryAspect {
         }
     }
 }
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct ValidationTimeout {
+    pub time_of_dispatch: SystemTime,
+    pub delay: Duration,
+}
+
+impl ValidationTimeout {
```

Comment:
I am seeing a couple other places where we are using an `Option<SystemTime, Duration>` as a timeout:
- https://github.com/holochain/holochain-rust/blob/f725e8f741e12ee966dc2d08cf0a3f0c97a8d3b7/crates/core/src/action.rs#L120
- https://github.com/holochain/holochain-rust/blob/f725e8f741e12ee966dc2d08cf0a3f0c97a8d3b7/crates/core/src/action.rs#L153
- https://github.com/holochain/holochain-rust/blob/f725e8f741e12ee966dc2d08cf0a3f0c97a8d3b7/crates/core/src/action.rs#L175
- https://github.com/holochain/holochain-rust/blob/f725e8f741e12ee966dc2d08cf0a3f0c97a8d3b7/crates/core/src/network/state.rs#L37
- https://github.com/holochain/holochain-rust/blob/f725e8f741e12ee966dc2d08cf0a3f0c97a8d3b7/crates/core/src/network/state.rs#L46

Would it make sense to have this refactor be a bit more general and work for all those cases?
.  Instead of creating this special case `ValidationTimeout`

Reply:
This struct could even become a more general Timeout struct. We already have a Timeout struct which is just a newtype wrapper around a Duration so this would be something else for internal use. Then we could use this struct in all those other places, which could DRY up a lot of repeated logic around checking if the timeout has occurred.

Reply:
> This struct could even become a more general Timeout struct. We already have a Timeout struct which is just a newtype wrapper around a Duration so this would be something else for internal use. Then we could use this struct in all those other places, which could DRY up a lot of repeated logic around checking if the timeout has occurred.

I'm going to save this for another PR I think